src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInMetricsListener.cs

```diff
@@ -51,16 +54,60 @@ public override void OnStopActivity(Activity activity, object payload)
                 return;
             }
 
+            string host;
+
+            if (context.Request.Host.Port is null or 80 or 443)
+            {
+                host = context.Request.Host.Host;
+            }
+            else
+            {
+                host = context.Request.Host.Host + ":" + context.Request.Host.Port;
```

Comment:
for a follow up - see if we avoid this string alloc, by using cache/other technique?